sonar-plugin-api/src/main/java/org/sonar/api/ce/posttask/PostProjectAnalysisTaskTester.java

```diff
@@ -0,0 +1,530 @@
+        }
+
+        @Override
+        public Project getProject() {
+          return project;
+        }
+
+        @Override
+        public QualityGate getQualityGate() {
+          return qualityGate;
+        }
+
+        @Override
+        public Date getDate() {
+          return date;
+        }
+
+        @Override
+        public String toString() {
+          return "ProjectAnalysis{" +
+            "ceTask=" + ceTask +
+            ", project=" + project +
+            ", date=" + date.getTime() +
+            ", qualityGate=" + qualityGate +
+            '}';
+        }
+      });
+
+  }
+
+  @Beta
+  public static final class CeTaskBuilder {
+    private static final String ID_CAN_NOT_BE_NULL = "id can not be null";
+    private static final String STATUS_CAN_NOT_BE_NULL = "status can not be null";
+
+    @CheckForNull
+    private String id;
+    @CheckForNull
+    private CeTask.Status status;
+
+    private CeTaskBuilder() {
+      // prevents instantiation outside PostProjectAnalysisTaskTester
+    }
+
+    public CeTaskBuilder setId(String id) {
+      this.id = requireNonNull(id, ID_CAN_NOT_BE_NULL);
+      return this;
+    }
+
+    public CeTaskBuilder setStatus(CeTask.Status status) {
+      this.status = requireNonNull(status, STATUS_CAN_NOT_BE_NULL);
+      return this;
+    }
+
+    public CeTask build() {
+      requireNonNull(id, ID_CAN_NOT_BE_NULL);
+      requireNonNull(status, STATUS_CAN_NOT_BE_NULL);
+      return new CeTask() {
+        @Override
+        public String getId() {
+          return id;
+        }
+
+        @Override
+        public Status getStatus() {
+          return status;
+        }
+
+        @Override
+        public String toString() {
+          return "CeTask{" +
+            "id='" + id + '\'' +
+            ", status=" + status +
+            '}';
+        }
+      };
+    }
+  }
+
+  @Beta
+  public static final class ProjectBuilder {
+    private static final String UUID_CAN_NOT_BE_NULL = "uuid can not be null";
+    private static final String KEY_CAN_NOT_BE_NULL = "key can not be null";
+    private static final String NAME_CAN_NOT_BE_NULL = "name can not be null";
+    private String uuid;
+    private String key;
+    private String name;
+
+    private ProjectBuilder() {
+      // prevents instantiation outside PostProjectAnalysisTaskTester
+    }
+
+    public ProjectBuilder setUuid(String uuid) {
+      this.uuid = requireNonNull(uuid, UUID_CAN_NOT_BE_NULL);
+      return this;
+    }
+
+    public ProjectBuilder setKey(String key) {
+      this.key = requireNonNull(key, KEY_CAN_NOT_BE_NULL);
+      return this;
+    }
+
+    public ProjectBuilder setName(String name) {
+      this.name = requireNonNull(name, NAME_CAN_NOT_BE_NULL);
+      return this;
+    }
+
+    public Project build() {
+      requireNonNull(uuid, UUID_CAN_NOT_BE_NULL);
+      requireNonNull(key, KEY_CAN_NOT_BE_NULL);
+      requireNonNull(name, NAME_CAN_NOT_BE_NULL);
+      return new Project() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 26 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1188)